### PR TITLE
convert varying_celestial_tranform schema ref to uri instead of tag

### PR DESCRIPTION
**Pull request by braingram**
asdf 3.0 will begin issuing a warning when a non-uri tag is encountered in a schema `$ref`. One such tag exists in the varying celestial transform schema:
https://github.com/DKISTDC/dkist/blob/main/dkist/io/asdf/resources/schemas/varying_celestial_transform-1.0.0.yaml#L33

This PR updates the schema replacing the tag with the corresponding and equivalent uri (`http://stsci.edu/schemas/asdf/transform/transform-1.2.0`). The uri used to replace the tag is identical to the uri that is resolved by the tag and should have no impact on the schema or code that uses the schema (aside from preventing the warning when asdf 3.0 is released).

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ Bug Fixes
 - Fix minor bugs for header slicing functionality and expand test coverage for edge-cases. (`#275 <https://github.com/DKISTDC/dkist/pull/275>`_)
 - Fixed inverse transform in `.VaryingCelestialTransformSlit2D`. Which fixes a bug in VISP WCSes. (`#285 <https://github.com/DKISTDC/dkist/pull/285>`_)
 - Fix a bug preventing the transfer of a single dataset with :meth:`~dkist.net.transfer_complete_datasets`. (`#288 <https://github.com/DKISTDC/dkist/pull/288>`_)
+- Update varying celestial transform schema ref to use a uri instead of a tag.(`#298 <https://github.com/DKISTDC/dkist/pull/298>`_)
```
Comment (Cadair) [inline CHANGELOG.rst:17]: Same here as #297 could you add this to the `changelog/` directory please?

File: dkist/io/asdf/resources/schemas/varying_celestial_transform-1.0.0.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ properties:
       - tag: "core/ndarray-1.0.0"
       - tag: "tag:stsci.edu:asdf/unit/quantity-1.1.0"
   projection:
-    $ref: "tag:stsci.edu:asdf/transform/transform-1.2.0"
+    $ref: "http://stsci.edu/schemas/asdf/transform/transform-1.2.0"
```
Comment (Cadair) [inline dkist/io/asdf/resources/schemas/varying_celestial_transform-1.0.0.yaml:33]: Can this be / should this be `asdf://`?

Comment (braingram) [reply dkist/io/asdf/resources/schemas/varying_celestial_transform-1.0.0.yaml:33]: Unfortunately this is still `http`:
https://github.com/asdf-format/asdf-transform-schemas/blob/82dd76605ee62c85cf9945feea3028924a49dc3d/resources/stsci.edu/schemas/transform-1.2.0.yaml#L4

File: CHANGELOG.rst (hunk 1 above)
Comment (braingram) [reply CHANGELOG.rst]: Thanks! I added [298.bugfix.rst](b732ee713d38666f75c58435c20ccad2a697b2d0). Should this be a 'bugfix' or something else?

**Review by Cadair — APPROVED** [review]
